flocker/node/script.py

```diff
@@ -304,6 +264,17 @@ def flocker_dataset_agent_main():
     loopback block device backend.  Later it will be capable of starting a
     dataset agent using any of the support dataset backends.
     """
+    # XXX This should use dynamic dispatch in the deployer_factory
+    # There should be only AgentScript, not ZFSAgentScript, and it should
+    # do the right thing for the configured backend. FLOC-1791.
+
+    options = DatasetAgentOptions()
+
+    return FlockerScriptRunner(
```

Comment:
Because the dataset_agent_main was called from nowhere (no tests, and this wasn't previously in a systemd/upstart file), this should change no test or user-facing functionality.